stream-tooling.md

```diff
@@ -52,10 +48,36 @@ There is also a second Koji tag, `coreos-release`, for packages which have been
 
 We maintain a git repository containing the rpm-ostree treefile and lockfiles. This could be [fedora-coreos-config](https://github.com/coreos/fedora-coreos-config). We have one branch for each stream, and no main branch.
 
-For the mechanical streams, a nightly job will run the compose from the corresponding yum repos and SCM refs. This job will output a lockfile for each CPU architecture. Those lockfiles will be committed to Git to preserve a record of the build's contents, and the builds will be pushed to the corresponding ostree refs.  The {bodhi-updates, branched} lockfile will also be PR'd to the {testing-devel, next-devel} branch, the latter only during the part of the cycle where next-devel is maintained.  We want to keep the development branches ready to release, so those PRs are not merged unless green.
+For the mechanical streams, a nightly job will run the compose from the corresponding yum repos and SCM refs. This job will output a lockfile for each CPU architecture. Those lockfiles will be committed to Git to preserve a record of the build's contents, and the builds will be pushed to the corresponding ostree refs.  The branched lockfile will also be PR'd to the {testing-devel, next-devel} branch, the latter only during the part of the cycle where next-devel is maintained.  We want to keep the development branches ready to release, so those PRs are not merged unless green.
 
 The lockfiles produced from the automatic snapshot will never be hand-modified, and in the next/testing/stable branches will never be modified at all except during promotions.  Instead, pins (to older NEVRAs) and updates (to newer ones) will be hand-maintained in the Git branches in a separate lockfile that overrides the autogenerated ones.  These overrides will be the major distinction between the mechanical refs and the "curated" (development/production) refs.  Each curated branch will have one override file, which can carry both CPU-architecture-independent and architecture-specific overrides.
 
+### Streams Lifecycle
+
+#### Regular State
+Although all the streams are built from scratch (there is no tangible relations between streams) here is how the RPMs flow towards stable:
+
+The **testing-devel** Stream is built daily with the content of the `fedora-update` repository. If the build successfully pass the tests,
+the resulting lockfiles are committed to the **testing-devel** branch of `fedora-coreos-config`.
+Any new RPM receive the `coreos-pool` koji tag.
+
+For a release (approximately every two weeks), someone manually triggers the `bump-lockfile` job to propagate the **testing** content to the **stable** branch.
+The **testing-devel** content is propagated to the **testing** and **next** streams. Theses images are built from the `coreos-pool` yum repository.
```

Comment:
Is it not `testing-devel` -> `next` -> `testing` -> `stable`? Reading this I feel you're saying that `testing-devel` goes directly to both ( `testing` <- `testing-devel` -> `next`).

Reply:
when Fedora is not branched (right now), testing and next are the same .

Looking at [testing commit history](https://github.com/coreos/fedora-coreos-config/commits/testing/) : it's at [testing devel e13ff7d](https://github.com/coreos/fedora-coreos-config/commit/e13ff7d3a6f743f2132153b8f078cbb95ccc8d62) 


Looking at [next commit history](https://github.com/coreos/fedora-coreos-config/commits/next/) : it's at [testing devel e13ff7d](https://github.com/coreos/fedora-coreos-config/commit/e13ff7d3a6f743f2132153b8f078cbb95ccc8d62)